cpp/src/arrow/python/builtin_convert.cc

```diff
@@ -687,7 +718,8 @@ std::shared_ptr<SeqConverter> GetConverter(const std::shared_ptr<DataType>& type
     case Type::DATE64:
       return std::make_shared<DateConverter>();
     case Type::TIMESTAMP:
-      return std::make_shared<TimestampConverter>();
+      return std::make_shared<TimestampConverter>(
+          static_cast<TimestampType*>(&*type)->unit());
```

Comment:
More consistent with other parts of the code to use `static_cast<const TimestampType&>(*type).unit()`